typeDefs.js

```diff
@@ -171,7 +170,8 @@ const typeDefs = gql`
     ): [Products]!
 
     getAllComments: [Comments]
-    getCommentsById(id: ID!): Comments
+    getCommentById(id: ID!): Comments
+    getAllCommentsByProduct(id: ID!): [Comments]
```

Comment:
Add exclamation mark to show that it cannot return `null` as a result and `null` as an element of the array

Reply:
done